build_wheel.sh

```diff
@@ -0,0 +1,176 @@
+
+# --------------------------------------------------------- #
+# main ()                                                   #
+# drives the entire build process                           #
+# Returns: 0 on success, non-zero if something goes wrong.  #
+# --------------------------------------------------------- #
+main() {
+  echo "Starting build ..."
+  pre_build_setup &&
+    build &&
+    post_build_test_and_teardown &&
+    echo "Build Successful ... "
+}
+
+# --------------------------------------------------------- #
+# pre_build_setup ()                                        #
+# performs necessary setup for building a YOLACT wheel      #
+# Returns: 0 on success, non-zero if something goes wrong.  #
+# --------------------------------------------------------- #
+pre_build_setup() {
+  echo "Pre-Build Setup ... "
+  __validate_files && __copy_files && __fix_requirements && __build_setup
+}
+
+# --------------------------------------------------------- #
+# build ()                                                  #
+# actual build happens here                                 #
+# Returns: 0 on success, non-zero if something goes wrong.  #
+# --------------------------------------------------------- #
+build() {
+  echo "Build ... "
+  python3 -m build
+  rm --recursive "${REPO_NAME}"
+}
+
+# --------------------------------------------------------- #
+# post_build_test_and_teardown ()                           #
+# installation check + cleanup unnecessary files            #
+# Returns: 0 on success, non-zero if something goes wrong.  #
+# --------------------------------------------------------- #
+post_build_test_and_teardown() {
+  echo "Post-Build Test and Teardowns ... "
+  __test_build
+  __teardown
+}
+
+__validate_files() {
+  echo "Validating files ... "
+  # venv
+  if [[ ! -e "$(command -v python3)" ]]; then
+    echo "Is python installed?" 1>&2
+    return 1
+  fi
+
+  # setup.py
+  if [[ ! -e "${SETUP_FILE}" ]]; then
+    echo "${SETUP_FILE} file not found" 1>&2
+    return 3
+  fi
+}
+
+__copy_files() {
+  echo "Copying files ... "
+
+  # create empty yolact dir, overwrite if exists
+  [[ -d ${REPO_NAME} ]] && rm --recursive ${REPO_NAME}
+  mkdir --parents "${REPO_NAME}"
+
+  for folder in "${NEEDED_FOLDERS[@]}"; do
+    cp --recursive "${folder}" "${REPO_NAME}"
+  done
+
+  echo "Copying ${REQUIREMENTS_FILE} ... "
+  cp "${REQUIREMENTS_FILE}" "${REPO_NAME}"
+
+  echo "Finding and Copying Python Files ... "
+  find . -maxdepth 1 -type f -name "*${PYTHON_EXTENSION}" -exec cp {} "${REPO_NAME}" \;
+
+}
+
+__fix_requirements() {
+  echo "Fixing ${REQUIREMENTS_FILE} ... "
+  sed -i '/^sparseml/d' "${REPO_NAME}/${REQUIREMENTS_FILE}"
+
+  yolact_file=".*yolact.py"
+  echo "Fixing file level imports ... "
+  find "${REPO_NAME}" -maxdepth 1 -type f -name "*${PYTHON_EXTENSION}" -print0 | while read -r -d $'\0' file; do
+    echo "file is $file"
+    if ! [[ "${file}" =~ $yolact_file ]]; then
```

Comment:
Ooo nice, now here's a spot where you really do want the double brackets. Cool to see those in action :+1:

Reply:
Though actually, this begs the question... why not use a different variant of `find` above?
```bash
find "${REPO_NAME}" -maxdepth 1 -type f -regex ".*${yolact_file}.*\.py"
```

Escaping the period in `.py` becomes tricky, I guess.